test/services/auth.service.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { BadRequestException, INestApplication } from '@nestjs/common';
+import { clearDB, createUser, createNestAppInstance } from '../test.helper';
+import { AuthService } from '@services/auth/auth.service';
+
+let nestApp: INestApplication;
+let service: AuthService;
+
+beforeEach(async () => {
+  await clearDB();
+});
+
+beforeAll(async () => {
+  nestApp = await createNestAppInstance();
+  service = nestApp.get<AuthService>(AuthService);
+});
+
+describe('Authenticate', () => {
+  it('should validate if the user password is accepted', async () => {
+    const userData = await createUser(nestApp, {});
+
+    const verifyUser = await service.validateUser(userData.email, 'password');
+
+    expect(verifyUser).toStrictEqual(userData);
+  });
+
+  it('should not validate the user if password is incorrect', async () => {
+    const userData = await createUser(nestApp, {});
+
+    const verifyUser = await service.validateUser(userData.email, 'passwordd');
+
+    expect(verifyUser).toBe(false);
+  });
+
+  it('should not be able to find the user', async () => {
+    expect(
+      async () => await service.validateUser('fakeEmail@email.io', 'passwordd'),
+    ).rejects.toThrowError(
+      new BadRequestException('fakeEmail@email.io does not exist'),
+    );
+  });
+
+  it('should generate a JWT token', async () => {
+    const result = {
+      access_token: '1234',
+    };
+
+    jest.spyOn(service, 'generateJWT').mockImplementation(async () => result);
```

Comment:
Is it possible to mock the `JwtService` `sign` method instead? 
That way we can guarantee the result of this function is as the result is described? 

With the current implementation we can update the `generateJWT` return and this test would still pass because we mocked the whole method 😂